typescript/sdk/src/ism/metadata/arbL2ToL1.ts

```diff
@@ -0,0 +1,169 @@
+    const ism = AbstractMessageIdAuthorizedIsm__factory.connect(
+      context.ism.address,
+      this.core.multiProvider.getSigner(context.message.parsed.destination),
+    );
+    const verified = await ism.isVerified(context.message.id);
+    if (verified) {
+      this.logger.debug(
+        'Message is already verified, relaying without metadata...',
+      );
+      return '0x';
+    }
+
+    // else build the metadata for outbox.executeTransaction call
+    const matchingL2Tx = context.dispatchTx.logs
+      .filter((log) => eqAddressEvm(log.address, context.hook.arbSys))
+      .map((log) => ArbSys.parseLog(log))
+      .find((log) => {
+        const calldata: string = log.args.data;
+        const messageIdHex = context.message.id.slice(2);
+        return calldata && calldata.includes(messageIdHex);
+      });
+
+    assert(matchingL2Tx, 'No matching L2ToL1Tx event found');
+    this.logger.debug({ matchingL2Tx }, 'Found matching L2ToL1Tx event');
+
+    if (matchingL2Tx) {
+      const l2ToL1TxEvent: L2ToL1TransactionEvent = {
+        ...matchingL2Tx.args,
+        caller: matchingL2Tx.args.caller,
+        destination: matchingL2Tx.args.destination,
+        hash: matchingL2Tx.args.hash,
+        position: matchingL2Tx.args.position,
+        arbBlockNum: matchingL2Tx.args.arbBlockNum,
+        ethBlockNum: matchingL2Tx.args.ethBlockNum,
+        timestamp: matchingL2Tx.args.timestamp,
+        callvalue: matchingL2Tx.args.callvalue,
+        data: matchingL2Tx.args.data,
+      };
+
+      const reader = new L2ToL1MessageReader(
+        this.core.multiProvider.getProvider('sepolia'),
+        l2ToL1TxEvent,
+      );
+
+      const baseProvider = new providers.JsonRpcProvider(
+        this.core.multiProvider.getChainMetadata(
+          'arbitrumsepolia',
+        ).rpcUrls[0].http,
+      );
+      const arbProvider = new ArbitrumProvider(baseProvider, {
+        name: 'arbitrum-sepolia',
+        chainId: 421614,
+      });
+
+      const status = await reader.status(arbProvider);
+      // need to wait for the challenge period to pass before relaying
+      if (!status) {
+        throw new Error(
+          `Arbitrum L2ToL1 message isn't ready for relay. Wait until the challenge period before relaying again.`,
+        );
+      }
+
+      const outboxProofResult =
+        (await reader.getOutboxProof(arbProvider)) ?? [];
+
+      // extract the proof key from MessageBatchProofInfo
+      const outboxProof: string[] =
+        'proof' in outboxProofResult
+          ? outboxProofResult.proof
+          : outboxProofResult ?? [];
+
+      const metadata: ArbL2ToL1Metadata = {
+        proof: outboxProof,
+        index: l2ToL1TxEvent.position,
+        l2Sender: l2ToL1TxEvent.caller,
+        to: l2ToL1TxEvent.destination,
+        l2Block: l2ToL1TxEvent.arbBlockNum,
+        l1Block: l2ToL1TxEvent.ethBlockNum,
+        l2Timestamp: l2ToL1TxEvent.timestamp,
+        data: l2ToL1TxEvent.data,
```

Comment:
can we just make the ArbL2ToL1Metadata the same type as the arbitrum SDK L2toL1TxEvent?

Reply:
ArbL2ToL1Metadata is the type for the input to the executeTransaction call on the L1 vs L2toL1TxEvent is the event on the L2, it's not perfectly overlapping

Reply:
added a comment to clarify

Reply:
done